File: Python/ValidPalindromII.py
def validPalindrome(s):
    left = 0
    right = len(s)-1

    def checkPalindrome(s,l,r):
        while l<r:
            if s[l] != s[r]:
                return False
            l+=1
            r-=1
        return True

    while left < right:
        if s[left]!=s[right]:
            skipL,skipR = checkPalindrome(s,left+1,right),\
                checkPalindrome(s,left,right-1)
            return skipL or skipR
        left +=1
        right -=1    
    return True

File: Javascript/ValidPalindromII.js
var validPalindrom = function(s){
    let left = 0
    let right = s.length-1
    
    var checkPalindrome = function(s,l,r){
        while(l<r){
            if(s[l]!==s[r]){
                return false
            }
            l+=1
            r-=1
        }
        return true
    }
    while(left<right){
        if(s[left]!==s[right]){
            let skipL = checkPalindrome(s,left+1,right)
            let skipR = checkPalindrome(s,left,right-1)
            return (skipL || skipR)
        }
        left += 1
        right -= 1
    }
    return true
}